src/protocol/nfs/mount/mod.rs

```diff
@@ -21,6 +21,31 @@ use null::mountproc3_null;
 use umnt::mountproc3_umnt;
 use umnt_all::mountproc3_umnt_all;
 
+/// Checks if a requested path matches an export path with proper path separator handling.
+///
+/// This function prevents incorrect matches like `/data2/file` matching `/data` export
+/// by ensuring that after the export name prefix, we either have the end of the string
+/// or a path separator (`/`).
+///
+/// # Arguments
+///
+/// * `requested_path` - The path being requested by the client
+/// * `export_name` - The export path to match against
+///
+/// # Returns
+///
+/// * `bool` - true if the requested path properly matches the export
+fn matches_export_path(requested_path: &str, export_name: &str) -> bool {
+    if requested_path == export_name {
+        true
+    } else if requested_path.starts_with(export_name) {
+        // Check that the character after the export name is a path separator
+        requested_path.chars().nth(export_name.len()) == Some('/')
+    } else {
+        false
+    }
```

Comment:
it can be done without if else

```suggestion
    requested_path == export_name
        || (requested_path.starts_with(export_name)
            && requested_path.chars().nth(export_name.len()) == Some('/'))
```